internal/provider/resource_model.go

```diff
@@ -110,9 +111,37 @@ func resourceModelCreate(ctx context.Context, d *schema.ResourceData, meta inter
 		return diag.FromErr(err)
 	}
 
-	// TODO: If cloud is blank, we should set it to the default (returned in modelInfo)
+	// compute the cloud information required
+	var cloudNameInput, cloudRegionInput string = "", ""
+	for _, c := range cloud {
+		cloudEntry := c.(map[string]interface{})
+		cloudNameInput = cloudEntry["name"].(string)
+		cloudRegionInput = cloudEntry["region"].(string)
```

Comment:
Is this part of the code dead?  `cloud` is set on `d` below only when the `cloudChanged` is true, and that happens only when either `cloudNameInput == ""` or `cloudRegionInput == ""`.